src/metabase/api/search.clj

```diff
@@ -456,6 +456,7 @@
                             (map t2.realize/realize)
                             (map to-toucan-instance)
                             (filter check-permissions-for-model)
+                            (map #(assoc % :can_write (mi/can-write? %)))
```

Comment:
@JesseSDevaney, adding this should help, I think. Now, running `api/search?archived=true`
Gives the normal response, but inside the :data key a result will look something like this:

```clojure
{:description nil,
 :archived true, ;; <----- even shows for archived true
 :collection_position nil,
 :table_id nil,
 :bookmark false,
 :can_write true, ;; <---- can_write for the result
 :database_id nil,
 :pk_ref nil,
 :name "Audit custom reports",
 :table_schema nil,
 :collection_authority_level nil,
 :updated_at nil,
 :moderated_status nil,
 :model_index_id nil,
 :model_id nil,
 :model_name nil,
 :id 113,
 :table_description nil,
 :dashboardcard_count nil,
 :context nil,
 :initial_sync_status nil,
 :table_name nil,
 :collection {:id 113, :name "Audit custom reports", :authority_level nil},
 :model "collection"}
```

Reply:
Thank you so much for this, Adam! It works perfectly :heart: